### PR TITLE
Migrate app to jupyter-lab2, fix token validation issue

### PR DESCRIPTION
**Pull request by jakubsy**

### CODE REVIEW

File: packages/neptune-extension/src/labextension/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { DisposableDelegate } from '@phosphor/disposable';
```
Comment (kierson) [inline packages/neptune-extension/src/labextension/index.tsx]: You've removed `phosphor` and this is not gonna build properly.

File: packages/neptune-extension/src/labextension/utils/notebook.ts [hunk 2]
```diff
@@ -56,21 +56,19 @@ class Notebook implements PlatformNotebook {
     return this.context.save();
   };
 
-  executeKernelCode(code: string) {
-    this.context.session.ready.then(() => {
-      (this.context.session.kernel as Kernel.IKernel).requestExecute({ code });
-    });
+  async executeKernelCode(code: string) {
```
[the remark this answers is not in the document]
Comment (kierson) [reply packages/neptune-extension/src/labextension/utils/notebook.ts]: Look here:

`packages/neptune-extension/src/types/platform.ts`

File: packages/neptune-extension/src/labextension/index.tsx (hunk 1 above)
Comment (jakubsy) [reply packages/neptune-extension/src/labextension/index.tsx]: I had it offline, forgot to add it to changelist.

File: packages/neptune-extension/src/labextension/utils/notebook.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakubsy) [reply packages/neptune-extension/src/labextension/utils/notebook.ts]: Ok, fixed

File: packages/neptune-extension/src/common/api/auth.ts [hunk 3]
```diff
@@ -81,4 +81,20 @@ export function parseApiToken(apiTokenStr: string): ApiTokenParsed | undefined {
...
+async function exchangeApiTokenWithTimeout(backend: BackendApi, apiToken: string) {
+  const accessToken = await Promise.race<Promise<NeptuneOauthToken>>([
+    new Promise((resolve, reject) =>
+      setTimeout(() => reject(new Error('Request timeout')), 5000)
+    ),
+    backend.exchangeApiToken({ xNeptuneApiToken: apiToken })
+  ]);
+
+  // handle success response with status 0 and no content
+  if (!accessToken) {
```
[the remark this answers is not in the document]
Comment (jakubsy) [reply packages/neptune-extension/src/common/api/auth.ts]: ok

**Pull request merged**